src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Generator;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with replace identifiers
+ *     [
+ *         '_embedded' => [
+ *             'articles' => '__articles__',
+ *         ],
+ *     ],
+ *     // array of generators with replace identifier used as key
+ *     [
+ *         '__articles__' => (function (): \Generator { // any method or function returning a Generator
+ *              yield ['title' => 'Article 1'];
+ *              yield ['title' => 'Article 2'];
+ *              yield ['title' => 'Article 3'];
+ *         })(),
+ *     ]
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    protected int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[] $structure
+     * @param array<string, Generator<int|string, mixed> $generics
```

Comment:
Do we need to force generators there ? the code only cares about Traversable (or even iterable). Generators are not the only way to implement an iterator and this looks like an arbitrary restriction here.

Reply:
The code would work with `iterable` but Generators are recommended as that is the main thing why you can make the response performant and low on resources. If you just put things again into array or a collection it will not benifit a lot of the Response.

But I understand to allow iterable for better DX and will adopt it. Because there are already some issue in [Doctrine blocking toIterable](https://github.com/doctrine/orm/issues/9460) in some cases and creating different response object would make the DX not good.

Reply:
Update that iterable are fine but still the docs concentrate on the Generators as that will be in my case the most usecase for this type of response.